### DISCUSSION TITLE
2022-31: Databend 的工程效率实践

### DISCUSSION THREAD

**Discussion by giscus** [Xuanwo/blog#374] [Announcements]
# 2022-31: Databend 的工程效率实践

An infrastructure engineer, focused on distributed storage system

https://xuanwo.io/reports/2022-31/

**Comment by STRRL** [comment]
话说关于 GitHub Action 中的 flaky test 之类的治理, 有什么推荐的方法或者工具吗? 😍

**Reply by Xuanwo** [reply]
其实文章草稿里面本来有关于不稳定测试的内容，不过不是很系统，最后成文的时候删掉了。

Databend 现在的做法是这样： https://github.com/datafuselabs/databend/discussions/6595

基本上，我们把 flaky 对应的 test 视作 bug 来处理：

- 发现不稳定测试就创建一个 issue，如果持续不稳定就先把这个 case 先 disable
- 积极修复这些 bug，在一些特别的时间节点（比如大版本发布）还会做批量的 cleanup

**Reply by STRRL** [reply]
了解!

> 发现不稳定测试就创建一个 issue，如果持续不稳定就先把这个 case 先 disable

目前这个阶段是主要是人力去推吗? 或者说有木有啥比较好的工具可以统计 GitHub Action 失败情况的? OwO

**Reply by Xuanwo** [reply]
嗯，就是很简单的发现一个开一个，自动化搞出来一大堆不会有人看的。

自动化要配合实际需求落地：先判断要不要，然后再考虑怎么做。以不稳定测试这种场景，我觉得不是很有必要：测试失败率是 3% 还是 2% 意义很大吗？

**Reply by STRRL** [reply]
主要是记录一下，客观的观测一下当大家觉得它是 flaky teat 的时候，它已经失败过多少次 or 失败率是多少了

**Reply by zhyon404** [reply]
我见过做的比较好的是k8s,用的是https://github.com/GoogleCloudPlatform/testgrid
但是应该不能和Action集成

**Reply by zhyon404** [reply]
底层有基于bigquery做统计，识别flaky

**Reply by STRRL** [reply]
嗯 之前了解过一点 testgrid 但是它没法 slefhost. 而且自动识别 flaky 感觉也比较玄学, 不太倾向于用.

**Reply by zhyon404** [reply]
https://github.com/kubernetes/test-infra/blob/master/metrics/configs/flakes-daily-config.yaml
自动识别 flaky应该是用sql实现的，能看到怎么算的。（有点复杂，我没仔细看过
